cluster-autoscaler/cloudprovider/aws/ec2_instance_types/gen.go

```diff
@@ -0,0 +1,176 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// +build ignore
+
+package main
+
+import (
+	"encoding/json"
+	"html/template"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws/endpoints"
+	"github.com/golang/glog"
+)
+
+type response struct {
+	Products map[string]product `json:"products"`
+}
+
+type product struct {
+	Attributes productAttributes `json:"attributes"`
+}
+
+type productAttributes struct {
+	InstanceType string `json:"instanceType"`
+	VCPU         string `json:"vcpu"`
+	Memory       string `json:"memory"`
+}
+
+type instanceType struct {
+	InstanceType string
+	VCPU         int64
+	Memory       int64
+}
+
+var packageTemplate = template.Must(template.New("").Parse(`/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// This file was generated by go generate; DO NOT EDIT
+
+package aws
+
+type instanceType struct {
+	InstanceType string
+	VCPU         int64
+	MemoryMb     int64
+}
+
+// InstanceTypes is a map of ec2 resources
+var InstanceTypes = map[string]*instanceType{
+{{- range .InstanceTypes }}
+	"{{ .InstanceType }}": {
+		InstanceType: "{{ .InstanceType }}",
+		VCPU:         {{ .VCPU }},
+		MemoryMb:     {{ .Memory }},
+	},
+{{- end }}
+}
+`))
+
+func main() {
+	instanceTypes := make(map[string]*instanceType)
```

Comment:
Sorry If I'm not following you correctly but for me, it seems like we are overwriting instance types over and over again for every region into this map?

Perhaps we should make the keys of the `instanceTypes` map region-aware, or we end up with throwing away almost all the instance type details provided but for the last region(which was `us-gov-west-1` for me in the example below):

```
$ go run ec2_instance_types/gen.go -logtostderr
I0705 17:15:01.732876   21476 gen.go:106] fetching https://pricing.us-east-1.amazonaws.com/offers/v1.0/aws/AmazonEC2/current/ap-northeast-1/index.json
I0705 17:15:02.711280   21476 gen.go:106] fetching https://pricing.us-east-1.amazonaws.com/offers/v1.0/aws/AmazonEC2/current/us-west-2/index.json
I0705 17:15:03.444548   21476 gen.go:106] fetching https://pricing.us-east-1.amazonaws.com/offers/v1.0/aws/AmazonEC2/current/eu-west-2/index.json
I0705 17:15:03.820958   21476 gen.go:106] fetching https://pricing.us-east-1.amazonaws.com/offers/v1.0/aws/AmazonEC2/current/ap-south-1/index.json
I0705 17:15:04.286667   21476 gen.go:106] fetching https://pricing.us-east-1.amazonaws.com/offers/v1.0/aws/AmazonEC2/current/us-west-1/index.json
I0705 17:15:04.889539   21476 gen.go:106] fetching https://pricing.us-east-1.amazonaws.com/offers/v1.0/aws/AmazonEC2/current/ap-southeast-1/index.json
I0705 17:15:05.548524   21476 gen.go:106] fetching https://pricing.us-east-1.amazonaws.com/offers/v1.0/aws/AmazonEC2/current/eu-central-1/index.json
I0705 17:15:06.151498   21476 gen.go:106] fetching https://pricing.us-east-1.amazonaws.com/offers/v1.0/aws/AmazonEC2/current/ca-central-1/index.json
I0705 17:15:06.532968   21476 gen.go:106] fetching https://pricing.us-east-1.amazonaws.com/offers/v1.0/aws/AmazonEC2/current/us-east-1/index.json
I0705 17:15:07.211180   21476 gen.go:106] fetching https://pricing.us-east-1.amazonaws.com/offers/v1.0/aws/AmazonEC2/current/ap-northeast-2/index.json
I0705 17:15:07.710261   21476 gen.go:106] fetching https://pricing.us-east-1.amazonaws.com/offers/v1.0/aws/AmazonEC2/current/us-east-2/index.json
I0705 17:15:08.199422   21476 gen.go:106] fetching https://pricing.us-east-1.amazonaws.com/offers/v1.0/aws/AmazonEC2/current/eu-west-1/index.json
I0705 17:15:08.892445   21476 gen.go:106] fetching https://pricing.us-east-1.amazonaws.com/offers/v1.0/aws/AmazonEC2/current/ap-southeast-2/index.json
I0705 17:15:09.559583   21476 gen.go:106] fetching https://pricing.us-east-1.amazonaws.com/offers/v1.0/aws/AmazonEC2/current/sa-east-1/index.json
I0705 17:15:10.138488   21476 gen.go:106] fetching https://pricing.us-east-1.amazonaws.com/offers/v1.0/aws/AmazonEC2/current/cn-north-1/index.json
I0705 17:15:10.152407   21476 gen.go:124] Error unmarshaling https://pricing.us-east-1.amazonaws.com/offers/v1.0/aws/AmazonEC2/current/cn-north-1/index.json skipping...
I0705 17:15:10.152443   21476 gen.go:106] fetching https://pricing.us-east-1.amazonaws.com/offers/v1.0/aws/AmazonEC2/current/us-gov-west-1/index.json
```